src/frontend/src/view/team-aware-link.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { NavLink } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { AppState } from '../redux/reducer';
+import { selectValgtTeamId } from '../redux/team-velger-duck';
+
+interface HeaderStateProps {
+    valgtTeamId: string;
+}
+
+interface AnchorProps {
+    href: string;
+    valgtTeamId: string;
+    className?: string;
+    target?: string;
+    rel?: string ;
+    children?: any;
+}
+
+interface LinkProps {
+    to: string;
+    className: string;
+    children: any;
+    valgtTeamId: string;
+}
+
+function TeamAwareLink({to, className, children, valgtTeamId}: LinkProps) {
+    return (
+        <NavLink to={`${to}?team=${valgtTeamId}`} className={className}>{children}</NavLink>
+    );
+}
+
+function TeamAwareAnchor({href, valgtTeamId, children, className, target, rel}: AnchorProps) {
+    return (
+        <a href={`${href}?team=${valgtTeamId}`} className={className} target={target} rel={rel}>{children}</a>
+    );
+}
+
+const mapStateToProps = (state: AppState): HeaderStateProps => ({
+    valgtTeamId: selectValgtTeamId(state),
+});
+
+const connectedTeamAwareAnchor = connect(mapStateToProps)(TeamAwareAnchor);
+const connectedTeamAwareLink = connect(mapStateToProps)(TeamAwareLink);
+
+export { connectedTeamAwareAnchor as TeamAwareAnchor, connectedTeamAwareLink as TeamAwareLink};
```

Comment:
at du trenger å bruke `as` her er en codesmell. Er litt konotete å sette opp connect med typescript, men gjort riktig klarer den å infere typene. Man må bare definere OwnProps og StateProps hver for seg.

Reply:
Litt usikker på hvordan jeg skal få til det her uten å dele opp i to forskjellige filer? Kan jo droppe ```as```, men da må jeg eksportere med et annet navn en kompontens egentlige navn:

``` 
const connectedTeamAwareAnchor = connect(mapStateToProps)(TeamAwareAnchor);
const connectedTeamAwareLink = connect(mapStateToProps)(TeamAwareLink);

export { connectedTeamAwareAnchor, connectedTeamAwareLink };
```

Meget mulig det er en mer elegant vei, men jeg får ikke til å connecte to komponenter i selve exporten.

Reply:
jeg er med, leste feil. Det er problemet med at `as` er keyword i både javascript og typescript. Tenkte du gjorde casting mellom typer. Men ser nå det er renaming :) Og det er helt greit. :+1: